llama/sap.py

```diff
@@ -307,3 +307,34 @@ def generate_sap_data(today: datetime, invoices: List[dict]) -> str:
             sap_data += " "
             sap_data += "\n"
     return sap_data
+
+
+def generate_summary(
+    invoices: List[dict], data_file_name, control_file_name: str
```

Comment:
Since you've included type hints for the other two arguments, I would include one for `data_file_name` too.